From 8da888da80c760e5a052a0fdae248b9f540ba760 Mon Sep 17 00:00:00 2001
From: "Tan, Tee Min" <tee.min.tan@intel.com>
Date: Thu, 21 Nov 2019 11:08:18 +0800
Subject: [PATCH 3/9] HOTFIX: net: stmmac: add enhanced descriptor support to
 stmmac_vlan_insert()

Add support for Enhanced Tx Descriptor in stmmac_main.c for
stmmac_vlan_insert() function .

This patch eventually will squash to: ("net: stmmac: add Enhanced
Tx Descriptor support in main flow")

Signed-off-by: Tan, Tee Min <tee.min.tan@intel.com>
Signed-off-by: Voon Weifeng <weifeng.voon@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index a88a2232f18d..03cde1422abe 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -3566,7 +3566,11 @@ static bool stmmac_vlan_insert(struct stmmac_priv *priv, struct sk_buff *skb,
 
 	tag = skb_vlan_tag_get(skb);
 
-	p = tx_q->dma_tx + tx_q->cur_tx;
+	if (priv->enhanced_tx_desc)
+		p = &tx_q->dma_enhtx[tx_q->cur_tx].basic;
+	else
+		p = tx_q->dma_tx + tx_q->cur_tx;
+
 	if (stmmac_set_desc_vlan_tag(priv, p, tag, inner_tag, inner_type))
 		return false;
 
-- 
2.20.1

